bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ImageImpl.java

```diff
@@ -238,6 +245,33 @@ protected void initModel() {
         }
     }
 
+    /**
+     * Extracts the asset name from DAM resource
+     * 
+     * @return image name from DAM
+     */
+    protected String getImageNameFromDam() {
+        Resource damResource = request.getResourceResolver().getResource(fileReference);
+        if (damResource != null) {
+          Asset asset = damResource.adaptTo(Asset.class);
+          return getSeoFriendlyName(asset != null ? asset.getName() : "");
+        }
+          return "";
+	}
+    
+    /**
+     * Content editors can store DAM assets with white spaces in the name, this method makes
+     * the asset name SEO friendly and also makes it usable by the {@code AdaptiveImageServlet}
+     * 
+     * @param assetName
+     * @return name of the asset without extension
+     */
+    protected String getSeoFriendlyName(String assetName) {
+        assetName = StringUtils.isNotBlank(assetName)? assetName.trim().replaceAll(" ", "-").toLowerCase() : null;
```

Comment:
• Do we also need to strip non-supported characters here?
• I would remove the extension first before the string manipulation